### PR TITLE
NMS-15448: Only show acknowledged alarms in surveillance view

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-15448

### CODE REVIEW

File: features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        alarm.setServiceType(databasePopulator.getServiceTypeDao().findByName("ICMP"));
+        if (acknowledged) {
+            alarm.setAlarmAckUser("foobar");
+            alarm.setAlarmAckTime(new Date());
+        }
+        return alarm;
+    }
+
+    @Test
+    public void testSomething() {
```
Comment (synqotik) [inline features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java:109]: Perhaps rename to something like `testSurveillanceViewServiceReturnsOnlyUnacknowledgedAlarms` or similar.

File: features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        surveillanceViewService.setAlarmDao(databasePopulator.getAlarmDao());
+        final OnmsCategory cat1 = databasePopulator.getCategoryDao().findByName("Routers");
+        final OnmsCategory cat2 = databasePopulator.getCategoryDao().findByName("DEV_AC");
+
+        final List<OnmsAlarm> alarmsBefore = surveillanceViewService.getAlarmsForCategories(Sets.newHashSet(cat1), Sets.newHashSet(cat2));
+
+        Assert.assertEquals(1, alarmsBefore.size());
+
+        final OnmsEvent event1 = databasePopulator.buildEvent(distPoller);
+        event1.setEventCreateTime(new Date(1436881548292L));
```
Comment (synqotik) [inline features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java:123]: Maybe make the `Date` a constant, so it's clear it's just the same for every item.

File: features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        databasePopulator.getEventDao().save(event3);
+        databasePopulator.getEventDao().flush();
+
+        final OnmsAlarm alarm3 = buildAlarm(event3, false);
+        databasePopulator.getAlarmDao().save(alarm3);
+        databasePopulator.getAlarmDao().flush();
+
+        final List<OnmsAlarm> alarmsAfter = surveillanceViewService.getAlarmsForCategories(Sets.newHashSet(cat1), Sets.newHashSet(cat2));
+
+        Assert.assertEquals(3, alarmsAfter.size());
```
Comment (synqotik) [inline features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java:154]: Consider adding a comment (either inline or for the method) about this result, e.g. "There should be 3 unacknowledged alarms: the initial one from the databasePopulator, plus 2 of the 3 added in the test"

Comment (RangerRick) [reply features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java:154]: yeah, we should consider it a code smell to use an assert that doesn't use the 3-argument version (or 2, for `assertTrue`/`assertFalse`)

File: features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java (hunk 1 above)
Comment (christianpape) [reply features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java:109]: Oh my god. I forgot to rename the method. I'm sorry.

File: features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java (hunk 2 above)
Comment (christianpape) [reply features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java:123]: done.

File: features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java (hunk 3 above)
Comment (christianpape) [reply features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java]: done

**Review by synqotik — APPROVED** [review]
Looks good, thanks!

**Review by RangerRick — APPROVED** [review]

**Pull request merged**